notifications_utils/clients/zendesk/zendesk_client.py

```diff
@@ -34,6 +71,73 @@ def send_ticket_to_zendesk(self, ticket):
 
         current_app.logger.info(f"Zendesk create ticket {ticket_id} succeeded")
 
+    def _upload_attachment(self, attachment: NotifySupportTicketAttachment):
+        query_params = {"filename": attachment.filename}
+
+        upload_url = self.ZENDESK_UPLOAD_FILE_URL + "?" + urlencode(query_params)
+
+        response = requests.post(
+            upload_url,
+            headers={"Content-Type": attachment.content_type},
+            data=attachment.filedata,
+            auth=(f"{self.NOTIFY_ZENDESK_EMAIL}/token", self.api_key),
+        )
+
+        if response.status_code != 201:
+            current_app.logger.error(
+                f"Zendesk upload attachment request failed with {response.status_code} '{response.json()}'"
+            )
+            raise ZendeskError(response)
+
+        upload_token = response.json()["upload"]["token"]
+
+        current_app.logger.info(f"Zendesk upload attachment `{attachment.filename}` succeeded")
+
+        return upload_token
+
+    def update_ticket(
+        self,
+        ticket_id: int,
+        comment: Optional[NotifySupportTicketComment] = None,
+        due_at: Optional[datetime.datetime] = None,
+        status: Optional[NotifySupportTicketStatus] = None,
+    ):
+        data = {"ticket": {}}
+
+        if comment:
+            data["ticket"]["comment"] = {
+                "body": comment.body,
+                "public": comment.public,
+            }
+
+            if comment.attachments:
+                data["ticket"]["comment"]["uploads"] = []
+                for attachment in comment.attachments:
+                    data["ticket"]["comment"]["uploads"].append(self._upload_attachment(attachment))
+
+        if due_at:
+            data["ticket"]["due_at"] = due_at.strftime(DATETIME_FORMAT_ISO8601)
```

Comment:
I was thinking it might still be good to set `due_at` to the normal date. Then if the task doesn't run for some reason, it will still bubble up in zendesk to be done manually. Just a bit of a failsafe. We might _not_ need it - but also costs us very little to have it here in case.

I suspect we'll have a cronitor alert on the celery task running the report so we'd get alerted elsewhere, just a backup to also set due_at. Thoughts?

Reply:
yeah think its fine to go here in that case - i'll guess i'll consider the implications more closely when you make the matching api PR to use this 👍